app/views/contributions/show.html.erb

```diff
@@ -7,8 +7,8 @@
     <p class="subtitle"><%= contribution.description %></p>
     <% if contribution.matched? %>
       <i class="far fa-check-circle has-text-primary is-size-5 has-text-weight-semibold"><span class="ml-1">Claimed</span></i>
-    <% else %>
-      <%= button_to "Claim This #{contribution.type}", new_contribution_claim_path(contribution), class: "button is-primary", method: :get %>
+    <% else  %>
+      <%= button_to "Claim This #{contribution.type}", new_contribution_claim_path(contribution), class: "button is-primary", disabled: (@peer_to_peer && !contribution.person.email), method: :get %>
```

Comment:
I have a preference for not having compound logic like this in the view. Pulling it out of the view makes it easier to write a test around it

Some suggestions on alternative places to put this:
* a helper method
* a method on contribution
* a presenter object, a new plan Ruby class or a `SimpleDelegator` class, that accepts a contribution and adds methods to it like `publicly_claimable?`

I'm happy to go into my thoughts on this in more detail if you'd like